pkg/compute/init.go

```diff
@@ -75,19 +127,143 @@ func (c *InitCommand) Exec(in io.Reader, out io.Writer) (err error) {
+		c.description, err = text.Input(out, "Description: ", in)
+		if err != nil {
+			return fmt.Errorf("error reading input: %w", err)
+		}
+	}
+
+	if c.author == "" {
+		var defaultEmail string
+		if email := c.Globals.File.Email; email != "" {
+			defaultEmail = fmt.Sprintf(" [%s]", email)
+		}
+
+		c.author, err = text.Input(out, fmt.Sprintf("Author:%s ", defaultEmail), in)
+		if err != nil {
+			return fmt.Errorf("error reading input %w", err)
+		}
+		if c.author == "" {
+			c.author = defaultEmail
+		}
+	}
+
+	if c.from == "" {
+		text.Output(out, "%s", text.Bold("Template:"))
+		for i, template := range defaultTemplates {
+			text.Output(out, "[%d] %s", i, template.Name)
+		}
+		template, err := text.Input(out, "Choose option or type URL: [1] ", in, validateTemplateOptionOrURL)
+		if err != nil {
+			return fmt.Errorf("error reading input %w", err)
+		}
+		if template == "" {
+			template = "1"
+		}
+		if i, err := strconv.Atoi(template); err == nil {
+			template = defaultTemplates[i].Path
+		}
+		c.from = template
+	}
+
+	if c.domain == "" {
+		mathRand.Seed(time.Now().UnixNano())
+		defaultDomain := fmt.Sprintf("%s.%s", petname.Generate(3, "-"), defaultTopLevelDomain)
+		c.domain, err = text.Input(out, fmt.Sprintf("Domain: [%s] ", defaultDomain), in, validateDomain)
+		if err != nil {
+			return fmt.Errorf("error reading input %w", err)
+		}
+		if c.domain == "" {
+			c.domain = defaultDomain
+		}
+	}
+
+	if c.backend == "" {
+		c.backend, err = text.Input(out, "Backend (originless, hostname or IP address): [originless] ", in, validateBackend)
+		if err != nil {
+			return fmt.Errorf("error reading input %w", err)
+		}
+		if c.backend == "" || c.backend == "originless" {
+			c.backend = "127.0.0.1"
+		}
+	}
+
+	text.Break(out)
+
+	if !c.Globals.Verbose() {
+		progress = text.NewQuietProgress(out)
+	}
+
+	progress.Step("Creating service...")
+	service, err := c.Globals.Client.CreateService(&fastly.CreateServiceInput{
+		Name:    c.name,
+		Type:    "wasm",
+		Comment: c.description,
+	})
+	if err != nil {
+		return fmt.Errorf("error creating service: %w", err)
+	}
+	undoStack.Push(func() error {
+		return c.Globals.Client.DeleteService(&fastly.DeleteServiceInput{
+			ID: service.ID,
+		})
+	})
+
+	progress.Step("Creating domain...")
+	_, err = c.Globals.Client.CreateDomain(&fastly.CreateDomainInput{
+		Service: service.ID,
+		Version: 1,
+		Name:    c.domain,
+	})
+	if err != nil {
+		return fmt.Errorf("error creating domain: %w", err)
+	}
+	undoStack.Push(func() error {
+		return c.Globals.Client.DeleteDomain(&fastly.DeleteDomainInput{
+			Service: service.ID,
+			Version: 1,
+			Name:    c.domain,
+		})
+	})
+
+	progress.Step("Creating backend...")
+	_, err = c.Globals.Client.CreateBackend(&fastly.CreateBackendInput{
+		Service: service.ID,
+		Version: 1,
+		Name:    c.backend,
+		Address: c.backend,
+	})
+	if err != nil {
+		return fmt.Errorf("error creating backend: %w", err)
+	}
+	undoStack.Push(func() error {
```

Comment:
If you want undo functions to return an error — which is sensible — then logging each non-nil error when executing the undo stack would probably be the best bet.

Reply:
Added error logging as part of d4b73e2